src/PowerShellEditorServices.Protocol/MessageProtocol/Channel/ChannelBase.cs

```diff
@@ -0,0 +1,118 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Serializers;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Channel
+{
+    /// <summary>
+    /// Defines a base implementation for servers and their clients over a
+    /// single kind of communication channel.
+    /// </summary>
+    public abstract class ChannelBase
```

Comment:
Yeah, thought it presumes that every channel can be created from input/output System.IO.Streams which may not be true in all cases.  If it turns out that's not so easy, I may need to find a way to abstract MessageReader/Writer into interfaces that can expose other implementation types.